test/util/test_runners/integrations_tests_runner.py

```diff
@@ -0,0 +1,138 @@
+"""
+
+# Builtins
+import sys
+import os
+
+# Third party
+import unittest
+import base64
+import requests
+from pathlib import Path
+import logging
+import subprocess as sb
+import shutil
+
+# Internals
+from test.test_integration_cli import CliIntegrationTest
+from conjur.endpoints import ConjurEndpoint
+from test.util.test_runners.integration_test_case import IntegrationTestCaseBase
+from test.util.test_runners.test_runner_args import TestRunnerArgs
+from test.util.test_runners.params import ClientParams, TestEnvironmentParams
+from test.util.test_path_provider import TestRunnerPathProvider
+from conjur.constants import *
+
+
+def main():
+    args = TestRunnerArgs.create_from_args()
+
+    runner = TestRunner(args)
+    runner.start_env()
+    runner.run_tests(test_modules=[CliIntegrationTest])
+
+
+class TestRunner:  # pragma: no cover
+
+    def __init__(self, args: TestRunnerArgs):
+        self.runner_args = args
+        path_provider = TestRunnerPathProvider(helpers_file_dir=args.files_folder)
+        self.api_key = self.__get_api_key()
+        self.env_params = TestEnvironmentParams(args.cli_to_test_path, args.invoke_cli_as_process, path_provider)
+        self.client_params = ClientParams(self.runner_args.hostname, self.runner_args.username,
+                                          self.runner_args.account, self.api_key)
+
+    def run_tests(self, test_modules: list):
+        """
+        @param test_modules: the tests classes wanted to run
+        """
+        res = []
+        for test_module in test_modules:
+            res.append(self.__run_test_cases(test_module))
+        sys.exit(not all(res))
+
+    def start_env(self):
+        """
+        set up the test environment
+        Test environment setup will be in the following order:
+        1) Create a netrc file to provide login details
+        2) Run the CLI to initialize the CLI to get the certificate and conjurrc files
+        """
+
+        # copy netrc from home folder to test folder
+        path_provider = TestRunnerPathProvider.getInstance()
+        self.__create_netrc_file()
+        shutil.copy(path_provider.netrc_path,
+                    os.path.join(path_provider.helpers_files_path, path_provider.netrc_file_name))
+
+        proc = sb.Popen(
+            [self.runner_args.working_cli_path_only_for_init, "init", "-a", f"{self.runner_args.account}", "-u",
+             f"{self.runner_args.hostname}",
+             "--force"],
+            stdin=sb.PIPE,
+            stdout=sb.PIPE,
+            stderr=sb.PIPE)
+        output = proc.communicate(input=b'yes\n', timeout=10)[0]
+        if proc.returncode != 0:
+            raise RuntimeError(f"Error Init: {output}")
+        # copy conjurrc from home folder to test folder
+        shutil.copy(path_provider.conjurrc_path, path_provider.test_conjurrc_file_path)
+
+    def __create_netrc_file(self):
+        password = self.api_key
+        home_path = str(Path.home())
```

Comment:
isn't this controlled by the user inputting `--working_cli` to tell you where they want these files to be?

Reply:
this was before the login command was supported